File: windows-src/Flipqlo/Engine/DesignTokens.cs
using System;
using System.Windows;
using System.Windows.Media;

namespace Flipqlo.Engine;

/// <summary>
/// Platform-agnostic design constants derived from shared/design-tokens.json.
/// Defined as code constants for zero-allocation access during rendering.
/// </summary>
public static class DesignTokens
{
    // ── Colors ──────────────────────────────────────────────────────────
    public static readonly Color Background      = ColorFrom(0xFF0A0A0A);
    public static readonly Color CardFace         = ColorFrom(0xFF1C1C1C);
    public static readonly Color CardHighlight    = ColorFrom(0xFF242424);
    public static readonly Color DigitColor       = ColorFrom(0xFFD8D8D8);
    public static readonly Color DividerLine      = ColorFrom(0xFF0F0F0F);
    public static readonly Color DividerShadow    = ColorFrom(0xFF000000);
    public static readonly Color ColonColor       = ColorFrom(0xFF3A3A3A);
    public static readonly Color FlapShadowColor  = ColorFrom(0x59000000); // ~35% alpha

    // ── Pre-built frozen brushes ────────────────────────────────────────
    public static readonly Brush BackgroundBrush  = Freeze(new SolidColorBrush(Background));
    public static readonly Brush CardFaceBrush    = Freeze(new SolidColorBrush(CardFace));
    public static readonly Brush CardHighlightBrush = Freeze(new SolidColorBrush(CardHighlight));
    public static readonly Brush DigitBrush       = Freeze(new SolidColorBrush(DigitColor));
    public static readonly Brush ColonBrush       = Freeze(new SolidColorBrush(ColonColor));
    public static readonly Brush FlapShadowBrush  = Freeze(new SolidColorBrush(FlapShadowColor));
    public static readonly Pen DividerPen         = FreezePen(new Pen(new SolidColorBrush(DividerLine), 2.0));
    public static readonly Pen DividerShadowPen   = FreezePen(new Pen(new SolidColorBrush(DividerShadow), 1.0));

    // ── Animation ───────────────────────────────────────────────────────
    public const double FlipDurationMs       = 600.0;
    public const double FlipTopPhaseRatio    = 0.5;
    public const double TickIntervalMs       = 250.0;
    public const double FlapShadowMaxAlpha   = 0.35;

    // ── Layout (ratios of card height unless noted) ─────────────────────
    public const double CardCornerRadiusPct  = 4.0;   // % of card height
    public const double CardAspectRatio      = 0.75;  // width / height
    public const double DigitToCardHeightPct = 68.0;
    public const double ColonWidthToCardPct  = 25.0;
    public const double InterDigitGapPct     = 3.0;
    public const double GroupGapPct          = 6.0;
    public const double ClockToScreenPct     = 32.0;
    public const double VerticalCenterBias   = 0.5;

    // ── Helpers ─────────────────────────────────────────────────────────
    private static Color ColorFrom(uint argb)
    {
        return Color.FromArgb(
            (byte)(argb >> 24),
            (byte)(argb >> 16),
            (byte)(argb >> 8),
            (byte)(argb));
    }

    private static Brush Freeze(Brush b) { b.Freeze(); return b; }
    private static Pen FreezePen(Pen p)
    {
        if (p.Brush is Freezable fb) fb.Freeze();
        p.Freeze();
        return p;
    }
}

File: android-src/app/src/main/java/com/flipclo/screensaver/DesignTokens.kt
package com.fliqlo.screensaver

import android.graphics.Color

/**
 * Design constants matching the Windows version exactly.
 * Mirrors FliqloScr.Engine.DesignTokens from the Windows codebase.
 */
object DesignTokens {
    // ── Colors ──────────────────────────────────────────────────────────
    val background:      Int = 0xFF0A0A0A.toInt()
    val cardFace:        Int = 0xFF1C1C1C.toInt()
    val cardHighlight:   Int = 0xFF242424.toInt()
    val digitColor:      Int = 0xFFD8D8D8.toInt()
    val dividerLine:     Int = 0xFF0F0F0F.toInt()
    val dividerShadow:   Int = 0xFF000000.toInt()
    val colonColor:      Int = 0xFF3A3A3A.toInt()

    // ── Animation ───────────────────────────────────────────────────────
    const val flipDurationMs:     Long   = 600L
    const val flipDurationMsF:    Float  = 600f
    const val tickIntervalMs:     Long   = 250L
    const val flapShadowMaxAlpha: Float  = 0.35f

    // ── Layout (ratios of card height unless noted) ─────────────────────
    const val cardCornerRadiusPct: Float = 0.04f   // fraction of card height
    const val cardAspectRatio:     Float = 0.75f   // width / height
    const val digitToCardHeight:   Float = 0.68f
    const val colonWidthToCard:    Float = 0.25f
    const val interDigitGapPct:    Float = 0.03f
    const val groupGapPct:         Float = 0.06f
    const val clockToScreenPct:    Float = 0.32f
    const val verticalCenterBias:  Float = 0.5f
    const val colonDotRadiusPct:   Float = 0.035f
}
